src/dc_receive_imf.rs

```diff
@@ -836,8 +836,8 @@ async fn add_parts(
 
                 if is_mdn || is_location_kml {
                     is_hidden = true;
-                    if state == MessageState::InFresh {
-                        state = MessageState::InNoticed;
+                    if state == MessageState::InFresh || state == MessageState::InNoticed {
```

Comment:
Why check for `InNoticed`? Does this check change anything, can hidden messages even become "noticed"?

Reply:
The thought in adding this extra check was that it's not easily visible that hidden messages can't become "noticed" and that we want to be prepared for the (maybe unlikely) case that the state can be "InNoticed".

However, just checking `if incoming` is probably the best because that's what we actually want to check.